### PR TITLE
Remove an invalid `content-length` in a gRPC streaming response.

### PR DESCRIPTION
**Pull request by ikhoon**
Motivation:

If a wrong content-length is set, the request/stream is reset by the Netty HTTP/2 codec. Most gRPC server call implmentations do not set a `content-length` to response headers. But some gRPC implementations such as Monix-gRPC send a `content-length` which was in the request content-length. I guess the runtime uses the incoming `Metadata` as the response `Metadata` itself.
Related: https://github.com/monix/monix-grpc/issues/43

Modifications:

- Remove the `content-length` in headers of the gRPC streaming response if set

Result:

- A custom content-length is automatically stripped when response streaming is used in `GrpcService`.

### CODE REVIEW

**Review by minwoox — APPROVED** [review]
Thanks, @ikhoon! 👍

File: grpc/src/main/java/com/linecorp/armeria/server/grpc/AbstractServerCall.java [hunk 1]
```diff
@@ -487,6 +488,15 @@ private void doSendHeaders(Metadata metadata) {
                     builder.set(GrpcHeaderNames.GRPC_ENCODING, compressor.getMessageEncoding());
                 }
                 MetadataUtil.fillHeaders(metadata, builder);
+
+                // Delete the custom content-length of the streaming response which might be set by wrongly
+                // implemented stubs such as Monix-gRPC. https://github.com/monix/monix-grpc/issues/43
+                // If a wrong content-length is set, an RST_STREAM error occurs at the Netty level.
+                // We don't need to care about the content-length of a unary call because it is eventually
+                // adjusted when the response is aggregated.
```
Comment (jrhee17) [inline grpc/src/main/java/com/linecorp/armeria/server/grpc/AbstractServerCall.java:496]: Verified chunked encoding is used for http1 as well, so sgtm 👍

**Review by jrhee17 — APPROVED** [review]
Looks good to me! Thanks @ikhoon ! 🙇 👍 🙇

**Comment by minwoox** [comment]
Thanks, @ikhoon and @jrhee17!

**Pull request merged**